plugins/quetz_sql_authenticator/quetz_sql_authenticator/api.py

```diff
@@ -136,5 +132,23 @@ def _delete(
             detail=f"User {username} not found",
         )
     db.delete(credentials)
-    db.commit()
+    _commit_and_tranform_errors(db)
     return username
+
+
+def _commit_and_tranform_errors(db):
+    """Commit changes to the database and transform errors to HTTP status codes."""
+    try:
+        db.commit()
+    except Exception as e:
+        logger.error(
+            f"""
+        quetz-sql-authenticator encountered the following error \
+        while trying to commit changes to the database:
+        {e}
+        """
+        )
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="An internal error occured.",
+        )
```

Comment:
Ok, with what we've discussed before, do we still need to safe guard the `db.commit()` call? I think not because:

1. When we raise an error inside a route, we send an internal server error response back to the client (same as in this function).
2. By raising the error, we get the entire traceback, not only the error message (better than this function).

So, it seems like there is no harm in raising a potential error here, which would make this function redundant.

Sorry for going back and forth on this topic. I didn't consider the implications when I first proposed this.

Reply:
Agreed. It seems like FastAPI or whatever part handles the error just returns "Internal Server Error" as HTTP and logs the error 👍🏼